wagtail/admin/mail.py

```diff
@@ -85,30 +101,298 @@ def send_notification(page_revision_id, notification, excluded_user_id):
+                        email_content = render_to_string(template_set['text'], context).strip()
+
+                    kwargs = {}
+                    if getattr(settings, 'WAGTAILADMIN_NOTIFICATION_USE_HTML', False):
+                        kwargs['html_message'] = render_to_string(template_set['html'], context)
+
+                    # Send email
+                    send_mail(email_subject, email_content, [recipient.email], connection=open_connection, **kwargs)
+                    sent_count += 1
+                except Exception:
+                    logger.exception(
+                        "Failed to send notification email '%s' to %s",
+                        email_subject, recipient.email
+                    )
+
+        return sent_count == len(recipients)
+
+    def send_notifications(self, template_set, context, recipients, **kwargs):
+        return self.send_emails(template_set, context, recipients, **kwargs)
+
+
+class BaseWorkflowStateEmailNotifier(EmailNotifier):
+    """A base EmailNotifier to send updates for WorkflowState events"""
+
+    def __init__(self):
+        super().__init__((WorkflowState,))
+
+    def get_context(self, workflow_state, **kwargs):
+        context = super().get_context(workflow_state, **kwargs)
+        context['page'] = workflow_state.page
+        context['workflow'] = workflow_state.workflow
+        return context
+
+
+class WorkflowStateApprovalEmailNotifier(BaseWorkflowStateEmailNotifier):
+    """An EmailNotifier to send updates for WorkflowState approval events"""
+
+    notification = 'approved'
+
+    def get_recipient_users(self, workflow_state, **kwargs):
+        triggering_user = kwargs.get('user', None)
+        recipients = {}
+        requested_by = workflow_state.requested_by
+        if requested_by != triggering_user:
+            recipients = {requested_by}
+
+        return recipients
+
+
+class WorkflowStateRejectionEmailNotifier(BaseWorkflowStateEmailNotifier):
+    """An EmailNotifier to send updates for WorkflowState rejection events"""
+
+    notification = 'rejected'
+
+    def get_recipient_users(self, workflow_state, **kwargs):
+        triggering_user = kwargs.get('user', None)
+        recipients = {}
+        requested_by = workflow_state.requested_by
+        if requested_by != triggering_user:
+            recipients = {requested_by}
+
+        return recipients
+
+
+class WorkflowStateSubmissionEmailNotifier(BaseWorkflowStateEmailNotifier):
+    """An EmailNotifier to send updates for WorkflowState submission events"""
+
+    notification = 'submitted'
+
+    def get_recipient_users(self, workflow_state, **kwargs):
+        triggering_user = kwargs.get('user', None)
+        recipients = {}
+        include_superusers = getattr(settings, 'WAGTAILADMIN_NOTIFICATION_INCLUDE_SUPERUSERS', True)
+        if include_superusers:
+            recipients = get_user_model().objects.filter(is_superuser=True)
+            if triggering_user:
```

Comment:
Ah, just one more... this should be outside the `if include_superusers`

Reply:
Thanks, done!